contracts/Strategy.sol

```diff
@@ -119,24 +176,30 @@ contract Strategy is BaseStrategy {
 
         uint256 beforeWant = balanceOfWant();
 
-        collectTradingFees();
+        _collectTradingFees();
         // claim beets
-        claimRewards();
-        // consolidate % to stake and unstake
-        consolidate();
-        // sell the % not staking
-        sellRewards();
+        _claimRewards();
+        // this would allow finer control over harvesting to get credits in without selling
+        if (toggles.doSellRewards) {
+            _sellRewards();
+        }
 
         uint256 afterWant = balanceOfWant();
-
         _profit = afterWant.sub(beforeWant);
         if (_profit > _loss) {
             _profit = _profit.sub(_loss);
+            _debtPayment += _loss;
             _loss = 0;
         } else {
             _loss = _loss.sub(_profit);
+            _debtPayment += _profit;
             _profit = 0;
         }
+
+        if (toggles.checkDebtOutstanding) {
+            // final check to make sure accounting is correct
+            require(_debtOutstanding == _debtPayment.add(_loss));
```

Comment:
this check needs to be in liquidatePosition, not prepareReturn

Reply:
Solved